pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -43,14 +43,22 @@ let
   );
 
   forceRust = features.rust or false;
-  kernelSupportsRust = lib.versionAtLeast version "6.7";
+  kernelSupportsRust = (
+    lib.versionAtLeast version "6.12"
+    && (
+      stdenv.hostPlatform.isx86_64
+      || stdenv.hostPlatform.isLoongArch64
+      || stdenv.hostPlatform.isAarch64
+      || (stdenv.hostPlatform.isRiscV64 && !stdenv.cc.isGNU)
```

Comment:
Would be nice to recheck if this still applies maybe

Reply:
It does, it even got restricted to clang (i.e. non-GCC) builds upstream now.

Reply:
This is giga nitpicky, but if you have a link to where this is defined upstream, it would be very cool to link to that here.

Reply:
I was more thinking of https://web.git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/rust/arch-support.rst really.

Reply:
fwiw, that table apparently not properly updated anyway, it still restricts ARM64 to little-endian only, even though that restriction has been gone since 6.12.

I'll try and get around to updating the link in the afternoon.

Reply:
Sadly no available shortlink for this.